code/game/objects/items/storage/holsters.dm

```diff
@@ -74,13 +83,39 @@
 /obj/item/storage/holster/update_icon()
 	. = ..() //calls update_icon_state to change the icon/item state
 	var/mob/user = loc
-	if (!istype(user))
+	if(!istype(user))
 		return
 	user.update_inv_back()
 	user.update_inv_belt()
 	user.update_inv_s_store()
 
-//Will only draw the specific holstered item, not ammo etc.
+///We do not want to use regular update_icon as it's called for every item inserted. Not worth the icon math.
+/obj/item/storage/holster/proc/update_gun_icon()
```

Comment:
Minor note with tha above change, when this is generic to all holsters, we shouldn't reference gun in proc/var names, since holsters aren't always for guns.